packages/yew/src/virtual_dom/renderable.rs

```diff
@@ -0,0 +1,205 @@
+
+    /// Converts this type into a [`Html`].
+    fn into_html(self) -> Html
+    where
+        Self: Sized,
+    {
+        self.to_html()
+    }
+}
+
+// Implementations for common data types.
+
+impl<T> Renderable for Option<T>
+where
+    T: Renderable,
+{
+    #[inline(always)]
+    fn to_html(&self) -> Html {
+        self.as_ref().map(Renderable::to_html).unwrap_or_default()
+    }
+
+    #[inline(always)]
+    fn into_html(self) -> Html {
+        self.map(Renderable::into_html).unwrap_or_default()
+    }
+}
+
+impl<T> Renderable for Vec<T>
+where
+    T: Renderable,
+{
+    #[inline(always)]
+    fn to_html(&self) -> Html {
+        Html::VList(VList::with_children(
+            self.iter().map(Renderable::to_html).collect(),
+            None,
+        ))
+    }
+
+    #[inline(always)]
+    fn into_html(self) -> Html {
+        Html::VList(VList::with_children(
+            self.into_iter().map(Renderable::into_html).collect(),
+            None,
+        ))
+    }
+}
+
+impl Renderable for Option<VNode> {
+    #[inline(always)]
+    fn to_html(&self) -> Html {
+        self.clone().into_html()
+    }
+
+    #[inline(always)]
+    fn into_html(self) -> Html {
+        self.unwrap_or_default()
+    }
+}
+
+impl Renderable for Vec<VNode> {
+    #[inline(always)]
+    fn to_html(&self) -> Html {
+        self.clone().into_html()
+    }
+
+    #[inline(always)]
+    fn into_html(self) -> Html {
+        Html::VList(VList::with_children(self, None))
+    }
+}
+
+impl Renderable for VText {
+    #[inline(always)]
+    fn to_html(&self) -> Html {
+        self.clone().into()
+    }
+
+    #[inline(always)]
+    fn into_html(self) -> Html {
+        Html::VText(self)
+    }
+}
+
+impl Renderable for VList {
+    #[inline(always)]
+    fn to_html(&self) -> Html {
+        self.clone().into()
+    }
+
+    #[inline(always)]
+    fn into_html(self) -> Html {
+        Html::VList(self)
+    }
+}
+
+impl Renderable for ChildrenRenderer<VNode> {
+    #[inline(always)]
+    fn to_html(&self) -> Html {
+        self.clone().into()
+    }
+
+    #[inline(always)]
+    fn into_html(self) -> Html {
+        self.into()
+    }
+}
+
+impl<T> Renderable for VChild<T>
+where
+    T: BaseComponent,
+{
+    #[inline(always)]
+    fn to_html(&self) -> Html {
+        self.clone().into()
+    }
+
+    #[inline(always)]
+    fn into_html(self) -> Html {
+        VNode::VComp(self.into())
+    }
+}
+
+impl Renderable for () {
+    #[inline(always)]
+    fn to_html(&self) -> Html {
+        VNode::default()
+    }
+
+    #[inline(always)]
+    fn into_html(self) -> Html {
+        VNode::default()
+    }
+}
+
+impl<T> Renderable for &'_ T
+where
+    T: Renderable,
+{
+    fn to_html(&self) -> Html {
+        (*self).to_html()
+    }
+}
+
+macro_rules! impl_renderable_via_display {
+    ($from_ty: ty) => {
+        impl Renderable for $from_ty {
+            #[inline(always)]
+            fn to_html(&self) -> Html {
+                Html::VText(self.into())
```

Comment:
Nit: using `.to_string` instead of `.into` should make it more readable

Reply:
VText's constructor relies on `Into<AttrValue>`.
Using `.into()` saves a clone for Rc'ed values like `AttrValue`.